File: common/net/cazzar/CoreLib/CoreMod.java
/**
 *
 */
package net.cazzar.corelib;

import cpw.mods.fml.relauncher.IFMLCallHook;
import cpw.mods.fml.relauncher.IFMLLoadingPlugin;
import net.cazzar.corelib.asm.McpMappings;
import net.cazzar.corelib.asm.SrgAccessTransformer;
import net.cazzar.corelib.lib.LogHelper;

import java.io.File;
import java.lang.reflect.Field;
import java.lang.reflect.Modifier;
import java.util.Map;
import java.util.logging.Level;

public class CoreMod implements IFMLLoadingPlugin, IFMLCallHook {
    private static final boolean runtimeDeobfuscationEnabled = true;
    private static final String deobfuscationFileName = null;
    private static final File mcLocation = null;

    public static String getDeobfuscationFileName() {
        return deobfuscationFileName;
    }

    @Override
    public String[] getLibraryRequestClass() {
        return null;
    }

    @Override
    public String[] getASMTransformerClass() {
        return new String[]{
                "net.cazzar.corelib.asm.CoreLibAccessTransformer",
                "net.cazzar.corelib.asm.SrgAccessTransformer"
        };
    }

    @Override
    public String getModContainerClass() {
        return "net.cazzar.corelib.ModContainer";
    }

    @Override
    public String getSetupClass() {
        return this.getClass().getCanonicalName();
    }

    @Override
    public void injectData(Map<String, Object> data) {
        for (String key : data.keySet()) {
            try {
                Field f = this.getClass().getDeclaredField(key);
                f.setAccessible(true);
                Field modifiers = f.getClass().getDeclaredField("modifiers");
                modifiers.setAccessible(true);
                modifiers.setInt(f, f.getModifiers() & ~Modifier.FINAL);
                f.set(null, data.get(key));
            } catch (NoSuchFieldException ignored) {
            } catch (IllegalAccessException e) {
                LogHelper.coreLog.log(Level.WARNING, e, "Unable to set field: %s", key);
            }
        }
    }

    @Override
    public Void call() throws Exception {
        McpMappings.instance();

        new SrgAccessTransformer("cazzarcorelib_at.cfg");
        return null;
    }
}

File: src/main/scala/net/cazzar/corelib/CoreMod.scala
package net.cazzar.corelib

import cpw.mods.fml.relauncher.{IFMLCallHook, IFMLLoadingPlugin}
import java.util
import java.io.File
import java.lang.reflect.Field

class CoreMod extends IFMLLoadingPlugin with IFMLCallHook {
  private var runtimeDeobfuscationEnabled: Boolean = true
  private var deobfuscationFileName: String = null
  private var mcLocation: File = null
  private final val FINGERPRINT: String = "B6:9D:73:36:FB:E4:C3:E9:72:79:EB:3E:E3:19:9F:00:9A:90:34:75".toLowerCase.replace(":", "")
  private var coremodLocation: File = null

  def getASMTransformerClass: Array[String] = Array()

  def getModContainerClass: String = "net.cazzar.corelib.ModContainer"

  def getSetupClass: String = getClass.getCanonicalName

  def getAccessTransformerClass: String = "net.cazzar.corelib.asm.CoreAccessTransformer"

  def call(): Void = {
    null
  }

  def injectData(data: util.Map[String, AnyRef]): Unit = {
    for (key <- data.keySet) {
      try {
        val f: Field = this.getClass.getDeclaredField(key)
        f.setAccessible(true)
        f.set(null, data.get(key))
      }
      catch {
        case ignored: Nothing => {
        }
        case e: Nothing => {
          LogHelper.coreLog.catching(e)
          LogHelper.coreLog.warn("Unable to set field: %s", key)
        }
      }
    }
  }
}
